plugins/formal-verification/src/org/jetbrains/kotlin/formver/conversion/StmtConversionVisitor.kt

```diff
@@ -154,12 +149,14 @@ class StmtConversionVisitor : FirVisitor<Exp, StmtConversionContext>() {
             return specialFunc.convertCall(getArgs(), data)
         }
 
-        val args = getArgs()
+        var args = getArgs()
         val symbol = functionCall.calleeReference.resolved!!.resolvedSymbol as FirNamedFunctionSymbol
         val calleeSig = data.add(symbol)
         val returnVar = data.newAnonVar(calleeSig.returnType)
         val returnExp = returnVar.toLocalVar()
         data.addDeclaration(returnVar.toLocalVarDecl())
+        // Convert arguments to the types expected by the function
+        args = args.zip(calleeSig.params).map { (arg, param) -> arg.withType(param.type) }
```

Comment:
I think it would make more sense to keep `args` a `val` and just use `getArgs().zip(...)` here.